test/parallel/test-assert.js

```diff
@@ -96,13 +96,6 @@ assert.throws(makeBlock(a.deepEqual, /a/i, /a/));
 assert.throws(makeBlock(a.deepEqual, /a/m, /a/));
 assert.throws(makeBlock(a.deepEqual, /a/igm, /a/im));
 
-{
-  const re1 = /a/;
-  re1.lastIndex = 3;
-  assert.throws(makeBlock(a.deepEqual, re1, /a/));
-}
```

Comment:
Instead of removing this test, would it make sense to change it to `assert.doesNotThrow`?